#### Data In Motion
# Pandas Weekly Challenge - Week 1
by Vaibhav Shaharwale

LinkedIn : https://www.linkedin.com/in/vaibhav-shaharwale-65460b177/


Challenge Questions.
1. Which was the most-ordered item?
2. For the most-ordered item, how many items were ordered?
3. What was the most ordered item in the choice_description column?
4. How many items were ordered in total?
5. Turn the item price into a float
6. How much was the revenue for the period in the dataset?
7. How many orders were made in the period?
8. What is the average revenue amount per order?
9. How many different items are sold?

In [1]:
import pandas as pd
data_import = pd.read_csv('https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv', sep='\t')
data_import.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

In [2]:
chipotle = data_import.copy()
chipotle.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


* the dtyep of item_price is object due to $ symbol.
* there are null values in choice_description column.

In [3]:
chipotle.isna().sum()

order_id                 0
quantity                 0
item_name                0
choice_description    1246
item_price               0
dtype: int64

##### Check duplicate entries

In [4]:
chipotle.duplicated().sum()

59

##### Remove Duplicate entries

In [5]:
chipotle.drop_duplicates(keep='first', inplace=True)

In [6]:
chipotle.shape

(4563, 5)

##### Fill null values in choice_description by "None".

In [7]:
chipotle['choice_description'].fillna('None', inplace=True)

In [8]:
chipotle.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                               None     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                               None     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

In [9]:
chipotle.quantity.value_counts()

1     4296
2      224
3       28
4       10
5        1
15       1
7        1
8        1
10       1
Name: quantity, dtype: int64

# 1. Which was the most-ordered item?

In [10]:
chipotle.item_name.value_counts().head()

Chicken Bowl           717
Chicken Burrito        546
Chips and Guacamole    474
Steak Burrito          365
Canned Soft Drink      290
Name: item_name, dtype: int64

#### Chicken Bowl is the most ordered item with 717 orders, 
#### followed by Chicken Burrito and Chips and Guacamole with 546 and 474 orders resp.

# 2. For the most-ordered item, how many items were ordered?

In [11]:
most_ordered_item = chipotle.groupby('item_name').quantity.sum()
most_ordered_item = most_ordered_item.to_frame()
most_ordered_item = most_ordered_item.reset_index()
most_ordered_item = most_ordered_item.set_index(pd.Index(range(len(most_ordered_item))))
most_ordered_item = most_ordered_item.sort_values(by='quantity', ascending=False)
most_ordered_item = most_ordered_item.set_index(pd.Index(range(len(most_ordered_item))))

# Top 5 most orderd Item with its quantity
most_ordered_item.head()

item_name  quantity
0         Chicken Bowl       752
1      Chicken Burrito       584
2  Chips and Guacamole       501
3        Steak Burrito       383
4    Canned Soft Drink       340

#### Chicken Bowl is the most ordered item, total 752 items were ordered.

# 3. What was the most ordered item in the choice_description column?

In [12]:
most_ordered_in_choice_des = chipotle.choice_description.value_counts()
most_ordered_in_choice_des = most_ordered_in_choice_des.reset_index()
most_ordered_in_choice_des.columns = ['choice_description', 'quantity']
most_ordered_in_choice_des = most_ordered_in_choice_des[most_ordered_in_choice_des['choice_description'] != 'None']

# Top 5 most ordered item in the choice_description
most_ordered_in_choice_des.head()

choice_description  quantity
1                                        [Diet Coke]       133
2                                             [Coke]       115
3                                           [Sprite]        77
4  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...        41
5  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...        38

#### Diet Coke is most ordered item in choice_description column.

# 4. How many items were ordered in total?

In [13]:
total_items = chipotle.quantity.sum()
total_items

4913

#### 4913 items were ordered in total.

# 5. Turn the item price into a float

In [19]:
chipotle['item_price'] = chipotle.item_price.str.replace('$', '')
chipotle['item_price'] = chipotle.item_price.astype('float64')
chipotle['item_price'].info()

<class 'pandas.core.series.Series'>
Int64Index: 4563 entries, 0 to 4621
Series name: item_price
Non-Null Count  Dtype  
--------------  -----  
4563 non-null   float64
dtypes: float64(1)
memory usage: 71.3 KB


# 6. How much was the revenue for the period in the dataset?

In [15]:
total_revenue = chipotle.item_price.sum()
total_revenue

34177.25

#### Total revenue for the period in the dataset is $34177.25

# 7. How many orders were made in the period?

In [16]:
total_orders = len(chipotle)
total_orders

4563

#### Total 4563 orders were made.

# 8. What is the average revenue amount per order?

In [17]:
avg_rev = total_revenue/total_orders
avg_rev = round(avg_rev, 3)
avg_rev

7.49

#### The average revenue amount per order is $7.49.

# 9. How many different items are sold?

In [18]:
items = chipotle.item_name.nunique()
items

50

#### Total 50 different item are sold.